nengo_spinnaker/operators/filter.py

```diff
@@ -19,252 +19,288 @@
+    into smaller vectors by transmitting each component with an appropriate
+    key; hence we can consider the entire operation of the filter component as
+    computing:
+
+    ..math:: c[z] = \mathbf{A} b[z]
+
+    Where **A** is the linear transform applied by the filter operator,
+    :math:`b[z]` is the filtered input vector and :math:`c[z]` is the nominal
+    output vector.
+
+    If **A** is of size :math:`m \times n` then *n* determines how many packets
+    each processing core (or group of processing cores) needs to receive and
+    *m* determines how many packets each processing core (or group of cores)
+    needs to transmit. To keep the number of packets received small we perform
+    column-wise partition of A such that:
+
+    ..math:: c[z] = \mathbf{A_1} b_1[z] + \mathbf{A_2} b_2[z]
+
+    Where :math:`\mathbf{A_x} b_x[z]` is the product computed by one set of
+    processing cores and :math:`c[z]` is the resultant vector as constructed by
+    any cores which receive packets from cores implementing the filter
+    operator. Note that the sum of products is computed by the receiving cores.
+    **A** and `b` are now partitioned such that **A** is of size :math:`m
+    \times (\frac{n}{2})` and `b` is of size :math:`\frac{n}{2}`; this reduces
+    the number of packets that need to be received by any group of cores
+    implementing the filter operator.
+
+    To reduce the number of packets which need to be transmitted by each core
+    we partition :math:`A_x` into rows such that:
+
+    ..math::
+        c =
+        \begin{pmatrix}
+          A_{1,1}~b_1 & + & A_{1,2}~b_2\\
+          A_{2,1}~b_1 & + & A_{2,2}~b_2
+        \end{pmatrix}
+
+    Where, in this example, :math:`A_{x,y}` is of size :math:`\frac{m}{2}
+    \times \frac{n}{2}`. Hence both the number of packets received and
+    transmitted by each core has been halved, and the number of
+    multiply-accumulates performed by each core has been quartered.  This
+    reduction in communication and computation in the filter operator is
+    achieved at the cost of requiring any operator with input `c` to receive
+    twice as many packets as previously (one set of packets for each
+    column-wise division) and to perform some additions.
     """
-    def __init__(self, size_in, n_cores_per_chip=None, n_chips=None):
+    def __init__(self, size_in):
         """Create a new parallel Filter.
 
         Parameters
         ----------
         size_in : int
             Width of the filter (length of any incoming signals).
-        n_cores_per_chip : int or None
-            Number of cores to use per chip to simulate the filter. If None
-            then a sensible value is chosen.
-        n_chips : int or None
-            Number of chips to use. If None then a sensible value is chosen.
         """
+        # Create as many groups as necessary to keep the size in of any group
+        # less than 128.
```

Comment:
It would be nice if the hand-waving was at least approximately explained in the comment, i.e. does this emerge from hand-waving related to some limit on incoming packets